modules/page-memory/src/test/java/org/apache/ignite/internal/pagememory/persistence/PersistentPageMemoryNoLoadTest.java

```diff
@@ -277,11 +309,47 @@ private static FilePageStoreManager createFilePageStoreManager(Path storagePath)
         return new FilePageStoreManager(log, "test", storagePath, new RandomAccessFileIoFactory(), PAGE_SIZE);
     }
 
-    private static void initGroupFilePageStores(FilePageStoreManager filePageStoreManager) throws Exception {
-        filePageStoreManager.initialize("Test", GRP_ID, PARTITION_ID + 1);
+    private static void initGroupFilePageStores(
+            FilePageStoreManager filePageStoreManager,
+            PartitionMetaManager partitionMetaManager,
+            CheckpointManager checkpointManager
+    ) throws Exception {
+        int partitions = PARTITION_ID + 1;
+
+        filePageStoreManager.initialize("Test", GRP_ID, partitions);
+
+        checkpointManager.checkpointTimeoutLock().checkpointReadLock();
+
+        try {
+            for (int i = 0; i < partitions; i++) {
+                FilePageStore filePageStore = filePageStoreManager.getStore(GRP_ID, i);
+
+                filePageStore.ensure();
+
+                GroupPartitionId groupPartitionId = new GroupPartitionId(GRP_ID, i);
 
-        for (FilePageStore filePageStore : filePageStoreManager.getStores(GRP_ID)) {
-            filePageStore.ensure();
+                CheckpointProgress currentProgress = checkpointManager.currentProgress();
+
+                PartitionMeta partitionMeta = partitionMetaManager.readOrCreateMeta(
+                        currentProgress == null ? null : currentProgress.id(),
+                        groupPartitionId,
+                        filePageStore
+                );
+
+                partitionMetaManager.addMeta(groupPartitionId, partitionMeta);
+
+                filePageStore.setPageAllocationListener(pageIdx -> {
+                    assert checkpointManager.checkpointTimeoutLock().checkpointLockIsHeldByThread();
+
+                    CheckpointProgress curr = checkpointManager.currentProgress();
```

Comment:
What does "currentProgress()" return if checkpoint is already finished?

Reply:
fix it